### PR TITLE
fix for decryption issues

### PR DESCRIPTION
**Pull request by prasad-maladi**
Observed issues while decrypting boolean values and nested encryption keys. 

Done the code changes to fix these issues

### CODE REVIEW

**Comment by koblas** [comment]
For this PR, I would be interested in getting a reference to the core `sops` encrypted and how it handles True/true.  I'm a little worried that `True` is not standard JSON.

**Comment by prasad-maladi** [comment]
> For this PR, I would be interested in getting a reference to the core `sops` encrypted and how it handles True/true. I'm a little worried that `True` is not standard JSON.


Cipher package is converting boolean value true as 'True' when it is encoding and it is giving the same as 'True' when it is decoding.  If I don't consider as 'True', it is failing with data integrity check and throwing "Hash mismatch error"

File: src/decode.ts [hunk 1]
```diff
@@ -187,7 +189,7 @@ export function decryptScalar(
     case "float":
       return parseFloat(cleartext);
     case "bool":
-      return cleartext === "true";
+      return cleartext === "True";
```
Comment (koblas) [inline src/decode.ts]: Can we change this to be case insensitive, matching `strconv.ParseBool(...)`

Comment (prasad-maladi) [reply src/decode.ts]: return cleartext.toLowerCase() === "true";   we can do like this for case insensitive

Comment (koblas) [reply src/decode.ts]: Work for me.

Comment (prasad-maladi) [reply src/decode.ts]: I updated the PR

**Comment by koblas** [comment]
Thanks.

Will publish 0.10.1 with this fix

**Comment by prasad-maladi** [comment]
thank you

**Pull request merged**